### ISSUE TITLE
Does not behave if moved from original location

### ISSUE THREAD

**Issue by gekidoslair** [soomla/unity3d-profile#31]
We have all of our 3rd party assets organized into a '3rd party' subfolder to keep our Unity projects from becoming a mess.

The Soomla libraries do not play nice with this kind of situation and are a pain to move. Facebook itself is another major fail in this regard, but it would be ideal if soomla could behave if moved into non-standard locations.

**Comment by refaelos** [comment]
@vedi can you answer here?

**Comment by refaelos** [comment]
@gekidoslair that's a good idea. Is the 3rd party folder under the Plugins folder? or under Plugins/Android (iOS) ?

**Comment by gekidoslair** [comment]
I'm looking to move the Assets/Soomla folder and the Assets/Plugins folder (both) into a folder called Assets/ThirdParty/Soomla folder.

The 'plugins' folder and 'resources' folders are special in the Unity build process - technically it doesn't matter where in the project structure they live - you can have multiple of them etc and Unity will combine / pack them all the same.

I realize that this gets trick with Post-Build scripts like you have setup, but definitely would be nice to have. The facebook SDK blows up in a similar manner, but ideally things should be setup to be (hopefully) able to handle / manage things irrregardless of where they are within the Unity project.

We have quite a few 3rd party assets in our project structure (some with plugins etc) that all work in this structure that I have defined above. 

The nice thing about this kind of structure is that it makes it easier to keep your Unity project clean - otherwise every assets / library you include in your project clutters up the root assets folder, your plugins folder becomes unmanageable etc.

We're trying to approach the problem slightly differently - keeping each unique asset in it's own sub-folder structure, under a main 'third party' folder.  The main project has quite a few separate plugins folders for example - and editor folders, resource folders etc - and Unity builds them fine.

Thanx for taking this into consideration.

Cheers,
Mike W

**Comment by refaelos** [comment]
Yeah you're totally right. We'll take a closer look into doing that.

If you want to create it, test it and PR your results it'll be great.

**Comment by EugeneButusov** [comment]
@gekidoslair, SOOMLA implemented this solution, all binaries will be placed in the separate folders starting from:
SOOMLA Core v1.2.0.
SOOMLA Profile v2.2.0
SOOMLA Store v1.9.0
SOOMLA Levelup v1.1.0.

Please, leave your feedback here, if this feature is still actual for you :smile:

**Comment by Whyser** [comment]
May I just jump in on this issue and ask, why does the `/Assets/Plugins/Soomla` folder actually live in the `/Assets/Plugins/` path? For example, as you've pointed out in the documentation, we need to copy the `Scripts`-folder of the Facebook SDK into the Plugin folder so that Soomla can access it's scripts. I see that you're steadily making it a lot easier for people to start using Soomla (removing Core and Events Prefab was nice!) and I feel that this step to move scripts from one folder to another shouldn't be necessary. Is this something you're thinking of changing? :)

**Comment by EugeneButusov** [comment]
@Whyser, the place selected for major SOOMLA files (`Assets/Plugins`) was selected by the reason, described there: http://docs.unity3d.com/Manual/ScriptCompileOrderFolders.html
Briefly, scripts located in the `Scripts` folder will be compiled firstly and then will be available for all solution classes.

**Comment by EugeneButusov** [comment]
@Whyser, could you please tell us if the current folder structure responds to your vision?
